yogstation/code/datums/components/fishingbonus.dm

```diff
@@ -4,22 +4,36 @@
 	var/mob/living/carbon/wearer
 
 /datum/component/fishingbonus/Initialize(fishing_bonus = 0)
-	if(!isclothing(parent))
+	if(!ismovable(parent))
 		return COMPONENT_INCOMPATIBLE
 	src.fishing_bonus = fishing_bonus
 	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/OnEquip)
 	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/OnUnequip)
+	RegisterSignal(parent, COMSIG_MOVABLE_BUCKLE, .proc/OnBuckle)
+	RegisterSignal(parent, COMSIG_MOVABLE_UNBUCKLE, .proc/OnUnbuckle)
 
 /datum/component/fishingbonus/proc/OnEquip(datum/source, mob/living/carbon/equipper, slot)
 	var/obj/item/parent_item = parent
-	var/obj/vehicle/ridden/lavaboat/parent_boat = parent
-	if(!parent_item || !parent_boat)
+	if(!parent_item)
 		return
-	if(parent_item.slot_flags == slotdefine2slotbit(slot) || parent_boat.post_buckle_mob(equipper))
+	if(parent_item.slot_flags == slotdefine2slotbit(slot))
 		equipper.fishing_power += fishing_bonus
 		wearer = equipper
 
 /datum/component/fishingbonus/proc/OnUnequip(datum/source, mob/living/carbon/equipper, slot)
 	if(wearer)
 		equipper.fishing_power -= fishing_bonus
 		wearer = null
+
+/datum/component/fishingbonus/proc/OnBuckle(datum/source, mob/living/carbon/M, force = FALSE)
+	var/obj/vehicle/ride = parent
+	if(!ride)
+		return
+	if(M in ride.occupants)
+		M.fishing_power += fishing_bonus
+		wearer = M
+
+/datum/component/fishingbonus/proc/OnUnbuckle(datum/source, mob/living/carbon/M, force = FALSE)
+	if(wearer)
```

Comment:
wearer is no longer appropriate as the variable name

Reply:
No

Reply:
Yes. If you're a boater then you're not "wearing" the boat. Fix it or I won't approve it.

Reply:
fine